index.d.ts

```diff
@@ -0,0 +1,267 @@
+import EventEmitter from 'events';
+import { PassThrough, Writable } from 'stream';
+
+type EventsType = {
+  PASSED: 'passed';
+  FAILED: 'failed';
+  SKIPPED: 'skipped';
+  STARTED: 'started';
+  FINISHED: 'finished';
+}
+
+type zUnitOptions = Partial<{
+  defaults: unknown;
+  initial: unknown;
+  runtime: unknown;
+  bequeathed: unknown;
+}>;
+
+declare class Runnable extends EventEmitter.EventEmitter {
+  constructor(name: string);
+
+  name: string;
+
+  description: string;
+}
+
+type zUnitSyntax = {
+  describe: typeof describe;
+  xdescribe: typeof xdescribe;
+  odescribe: typeof odescribe;
+  it: typeof it;
+  xit: typeof xit;
+  oit: typeof oit;
+  before: typeof before;
+  beforeEach: typeof beforeEach;
+  after: typeof after;
+  afterEach: typeof afterEach;
+  include: typeof include;
+}
+
+type OutcomesType = {
+  PASSED: 'passed';
+  FAILED: 'failed';
+  SKIPPED: 'skipped';
+}
+
+export const Events: EventsType;
+
+export class Harness extends EventEmitter.EventEmitter {
+  constructor(testable: Testable, initial?: zUnitOptions);
+
+  numberOfTests: number;
+
+  report: unknown;
+
+  run(reporter: unknown, runtime?: zUnitOptions): Promise<unknown>;
+}
+
+export class Hook extends Runnable {
+  constructor(name: string, fn: Function, initial?: zUnitOptions);
+
+  name: string;
+
+  run(propogatedOptions: zUnitOptions): Promise<void>;
+}
+
+export class HookSet {
+  addBefores(...additions: unknown[]): this;
+
+  addAfters(...additions: unknown[]): this;
+
+  runBefores(options: zUnitOptions): Promise<void>;
+
+  runAfters(options: zUnitOptions): Promise<void>;
+}
+
+export class Options {
+  constructor(options?: zUnitOptions);
+
+  defaults: unknown;
+
+  initial: unknown;
+
+  runtime: unknown;
+
+  bequeathed: unknown;
+
+  get(name: string): unknown;
+
+  export(): zUnitOptions;
+
+  apply(other: zUnitOptions): this;
```

Comment:
apply relies on duck typing - you can pass it an instance of an Option, or an inline map with optional 'defaults', 'initial', 'runtime' and 'bequeathed' sub-documents.

Reply:
If I understand you correctly, this has already been implemented, as I wrote `zUnitOptions` to be a type within a `Partial` utility type as so:
```ts
type zUnitOptions = Partial<{
  defaults: unknown;
  initial: unknown;
  runtime: unknown;
  bequeathed: unknown;
}>;
```
Which means each property is optional, so `apply` can be called as so:
```ts
apply({ defaults: ... })`
```
with just the `defaults` property, and so on for the other properties. Is this what you mean?

Reply:
Got it 👍